File: leetcode2588/solveLeetcode2588.cpp
// class Solution {
// public:
//     long long beautifulSubarrays(vector<int>& nums) 
//     {
//         const int n=nums.size();
//         unordered_map<int,int>hash;
//         hash[0]=1;

//         int prefix=0;
//         long long ans=0;
//         for(int i=0;i<n;++i)
//         {
//             prefix^=nums[i];

//             if(hash.count(prefix))ans+=hash[prefix];
//             ++hash[prefix];
//         }

//         return ans;
//     }
// };

File: DataStructure/leetcode2588/solveLeetcode2588.go
// func beautifulSubarrays(nums []int) int64 {
//     n:=len(nums)
//     f:=make(map[int]int)
//     f[0]=1

//     ans:=int64(0)
//     prefix:=0
//     for i:=0;i<n;i++{
//         prefix^=nums[i]

//         if val,ok:=f[prefix];ok{
//             ans+=int64(val)
//         }
//         f[prefix]++
//     }

//     return ans
// }